route/build/src/dfw_route.f90

```diff
@@ -1,135 +1,38 @@
 MODULE dfw_route_module
 
+! diffusive wave routing
+
 USE nrtype
-! data types
-USE dataTypes,   ONLY: STRFLX            ! fluxes in each reach
-USE dataTypes,   ONLY: STRSTA            ! state in each reach
-USE dataTypes,   ONLY: RCHTOPO           ! Network topology
-USE dataTypes,   ONLY: RCHPRP            ! Reach parameter
-USE dataTypes,   ONLY: dwRCH             ! dw specific state data structure
-USE dataTypes,   ONLY: subbasin_omp      ! mainstem+tributary data strucuture
-! global data
-USE public_var,  ONLY: iulog             ! i/o logical unit number
-USE public_var,  ONLY: realMissing       ! missing value for real number
-USE public_var,  ONLY: integerMissing    ! missing value for integer number
-USE public_var,  ONLY: qmodOption        ! qmod option (use 1==direct insertion)
-USE globalData,  ONLY: idxDW
-! subroutines: general
-USE water_balance, ONLY: comp_reach_wb  ! compute water balance error
-USE perf_mod,    ONLY: t_startf,t_stopf   ! timing start/stop
-USE model_utils, ONLY: handle_err
+USE dataTypes,     ONLY: STRFLX          ! fluxes in each reach
+USE dataTypes,     ONLY: STRSTA          ! state in each reach
+USE dataTypes,     ONLY: RCHTOPO         ! Network topology
+USE dataTypes,     ONLY: RCHPRP          ! Reach parameter
+USE dataTypes,     ONLY: dwRCH           ! dw specific state data structure
+USE public_var,    ONLY: iulog           ! i/o logical unit number
+USE public_var,    ONLY: realMissing     ! missing value for real number
+USE public_var,    ONLY: integerMissing  ! missing value for integer number
+USE public_var,    ONLY: qmodOption      ! qmod option (use 1==direct insertion)
+USE globalData,    ONLY: idxDW
+USE water_balance, ONLY: comp_reach_wb   ! compute water balance error
+USE base_route,    ONLY: base_route_rch
```

Comment:
Probably good to define a one line comment that this is the base class.

Reply:
adding comment: base (abstract) reach routing method class